sandbox/data_science/app_files/text_to_skills.py

```diff
@@ -0,0 +1,235 @@
+from gensim.models import LsiModel
+import pyLDAvis
+import pyLDAvis.gensim
+import warnings
+import pickle
+import itertools as it
+import sys
+
+from pprint import pprint
+
+from gensim.corpora import Dictionary, MmCorpus
+from gensim.models.ldamulticore import LdaMulticore
+from gensim.models import LsiModel
+from gensim.models.ldamulticore import LdaMulticore
+from gensim.models.ldamodel import LdaModel
+from gensim.models import Phrases
+from gensim.models.word2vec import LineSentence
+
+import gensim
+import en_core_web_sm
+import spacy
+nlp = spacy.load('en')
+
+from nltk.corpus import stopwords
+stopwords = stopwords.words('english')
+
+# Silence all warnings
+warnings.filterwarnings("ignore", '.*The binary mode.*')
+
+######################################################################
+###################### Handle command line args ######################
+######################################################################
+
+input_doc_name = sys.argv[1]
+with open(input_doc_name, 'r') as infile:
+    user_input_text = infile.read()
+
+num_skills_output = int(sys.argv[2])
+
+######################################################################
+############################ Set up model ############################
+######################################################################
+
+# load the finished bag-of-words corpus from disk
+trigram_bow_corpus_POS = MmCorpus('models/spacy_trigram_bow_corpus_all_POS.mm') # With POS preprocessing
+
+# load the finished dictionary from disk
+trigram_dictionary_POS = Dictionary.load('models/spacy_trigram_dict_all_POS.dict') # With POS preprocessing
+
+bigram_model = Phrases.load('models/spacy_bigram_model_all_PARSED_POS')
+trigram_model = Phrases.load('models/spacy_trigram_model_all_PARSED_POS')
+
+# load the finished LDA model from disk
+lda = LdaModel.load('models/spacy_lda_model_all_POS')
+
+topic_names = {1: u'(?)Large Tech Corps (NVIDIA, Splunk, Twitch)',
```

Comment:
So, topic modeling often returns topics that are not human-understandable; I've marked them with `(?)`. In our cluster of 25 topics, about three are topics that aren't oriented towards a particular type of job. Rather, they seem to be capturing particular patterns from the JDs of particular companies (i.e., the "benefits" section). For our purposes, those clusters are kind of garbage, but they appear to be somewhat unavoidable (perhaps unless we were to remove boilerplate language from the JDs as a preprocessing step, which seems like it'd be kind of tricky). 

Practically speaking, these garbage topics don't affect us much, because it's unlikely that a user's resume would match with one of them.